src/Hintman/App/Monad.hs

```diff
@@ -3,21 +3,80 @@
 module Hintman.App.Monad
        ( App (..)
        , AppEnv
+
+         -- * Runners
        , runAppAsHandler
+       , runAppLogIO
+       , runAppLogIO_
+
+         -- * Internal helpers
+       , runAppAsIO
        ) where
 
-import Control.Exception (try)
+import Control.Exception (catch, throwIO, try)
+import Control.Monad.Except (MonadError (..), liftEither)
+import Relude.Extra.Bifunctor (firstF)
 import Servant (Handler (..))
 
 import Hintman.App.Env (Env)
+import Hintman.App.Error (AppError, AppException (..), toHttpError)
+
 
+-- | 'Env' data type parameterized by 'App' monad
 type AppEnv = Env App
 
 -- | Main application monad.
 newtype App a = App
     { unApp :: ReaderT AppEnv IO a
-    } deriving (Functor, Applicative, Monad, MonadReader AppEnv, MonadIO)
+    } deriving newtype (Functor, Applicative, Monad, MonadReader AppEnv, MonadIO)
+
+instance MonadError AppError App where
+    throwError :: AppError -> App a
+    throwError = liftIO . throwIO . AppException
```

Comment:
Probably I'm wrong, but shouldn't we use our special `throwError` function from `App.Error`?

Reply:
No, it's vice versa actually. Our special function from `App.Error` uses this `throwError` as an implementation detail. So `throwError` from this instance tells how to throw `AppError` in the `App` monad. And `throwError` from `App.Error` takes `AppErrorType`, constructs `AppError` and then delegates to `MonadError.throwError`. Probably `throwError` from `App.Error` can be renamed, because after writing this stuff out I see how confusing it can be to have same names for different things...

Reply:
Oh, I see. Thanks for clarifications!